src/cls/IPM/ResourceProcessor/Default/Document.cls

```diff
@@ -609,23 +609,20 @@ Method OnGetUniqueName(Output pUniqueName)
 
 /// Returns a boolean of whether current file exists in the the routine database of the current namespace exists in current namespace's database
 /// Helper method to be used in derived classes' OnConfigureMappings() to skip creating unnecessary mapping
-ClassMethod FileExistsInCurrentNS(pFileName As %String) As %Boolean
+ClassMethod ResourceIsMappedToDefaultDB(pResourceName As %String) As %Boolean
 {
-	Set tSourceDB = ##class(%IPM.Utils.Module).GetRoutineDatabase($Namespace)
-	// Check if current file exists in the the routine database of the current namespace exists in current namespace('s database)
-	// If yes, skip creating unnecessary mappings
-	// (Spec,Dir=1,OrderBy=1,SystemFiles=1,Flat,NotStudio=0,ShowGenerated=0,Filter,RoundTime=0,Mapped=0)
-	// Set mapped=0 since we only want to check whether the file exists in the current routine DB
-	Set tResult = ##class(%Library.RoutineMgr).StudioOpenDialogFunc(pFileName_".*",1,1,1,,,,,,0)
-	Set tNeedExplicitMapping = 1
-	While tResult.%Next(.tSC) {
-		$$$ThrowOnError(tSC)
-		if ($ZConvert(tResult.%Get("Name"),"U") = $ZConvert(pFileName, "U")) {
-			Set tNeedExplicitMapping = 0
-		}
-	}
-	$$$ThrowOnError(tSC)
-	Return tNeedExplicitMapping
+	Set defaultDB = ##class(%IPM.Utils.Module).GetRoutineDatabaseDir($Namespace)
+  Set name = $Piece(pResourceName,".",1,*-1)
+  Set type = $ZConvert($Piece(pResourceName,".",*),"U")
+  If (type = "PKG") {
+    Set db = ##class(%SYS.Namespace).GetPackageDest(,name)
+  } ElseIf (type = "CLS") {
+    Set db = ##class(%SYS.Namespace).GetPackageDest(,$Piece(name,".",1,*-1))
+  } Else {
```

Comment:
Do we need a case for globals as well? Also, rather than grouping the specific use cases in the Document super class, would it make sense to have specific implementations in the respective sub classes?

Reply:
It was interesting looking back at the history of Beiming's work here - it started out as a separate implementations, then moved to a centralized (wrong in many ways) implementation.

In this case I focused on making it work for all of the existing callers, which all now pass an InternalName. Agreed that we should branch out to globals later; filed a separate issue for this: https://github.com/intersystems/ipm/issues/658

Reply:
That's fair - will cover refactoring in the followup task as well.